components/Tabs/Tabs.js

```diff
@@ -1,66 +1,84 @@
-
 class TabItem {
   constructor(element) {
-    // attach dom element to object. Example in Tabs class
+    this.element = element;// attach dom element to object. Example in Tabs class
+    this.classList = element.classList;
   }
 
   select() {
-    // should use classList
+    this.classList.add("Tabs__item-selected");// should use classList
   }
 
   deselect() {
-    // should use classList
+    this.classList.remove("Tabs__item-selected");// should use classList
   }
 }
 
 class TabLink {
-  constructor(element, parent) {
-    this.element;// attach dom element to object
-    this.tabs;// attach parent to object
-    this.tabItem;// assign this to the associated tab using the parent's "getTab" method by passing it the correct data
-    // reassign this.tabItem to be a new instance of TabItem, passing it this.tabItem
-    this.element.addEventListener('click', () => {
-      this.tabs.updateActive(this);
-      this.select();
-    });
-  };
+  constructor(element) {
+    this.element = element;// attach dom element to object
+  }
 
   select() {
-    // select this link
-    // select the associated tab
+    this.element.classList.add("Tabs__link-selected");// select this link
   }
 
   deselect() {
-    // deselect this link
-    // deselect the associated tab
+    this.element.classList.remove("Tabs__link-selected");// deselect this link
   }
 }
 
+// "Try to leverage the fact that the link and the item share some data,
+// and make the link ignorant of the item."
+
 class Tabs {
   constructor(element) {
     this.element = element;// attaches the dom node to the object as "this.element"
     this.links = element.querySelectorAll(".Tabs__link");
-    this.links = Array.from(this.links).map((link) => {
-      return new TabLink(link, this);
+    this.links = Array.from(this.links).map(link => {
+      return new TabLink(link);
     });
+    this.items = element.querySelectorAll(".Tabs__item");
+    this.items = Array.from(this.items).map(item => {
+      return new TabItem(item);
+    });
+    document.addEventListener("click", () => {
+      if (event.target.dataset.tab) {
+        const data = event.target.dataset.tab;
+        let tabLink = this.element.querySelector(`.Tabs__link[data-tab="${data}"]`);
+        tabLink = new TabLink(tabLink);
+        let tabItem = this.element.querySelector(`.Tabs__item[data-tab="${data}"]`);
+        tabItem = new TabItem(tabItem);
+        this.updateActive(tabLink, tabItem);
+        tabLink.select();
+        tabItem.select();
+      }
+    });
+
     this.activeLink = this.links[0];
+    this.activeItem = this.items[0];
     this.init();
   }
 
   init() {
-    // select the first link and tab upon ititialization
+    this.activeLink.select();// select the first link and tab upon ititialization
+    this.activeItem.select();
   }
 
-  updateActive(newActive) {
-    // deselect the old active link
-    // assign the new active link
+  updateActive(newActiveLink, newActiveItem) {
```

Comment:
Looks good and even if you just lift it like this the code will still be working and it'll be shorter.

but over all good job  keep up the good work :)

updateActive(newActive) {
     this.activeLink.deselect();
     this.activeLink = newActive;
   }